qcodes/utils/dataset/doNd.py

```diff
@@ -101,7 +106,14 @@ def do0d(
         The QCoDeS dataset.
     """
     meas = Measurement()
-    _register_parameters(meas, param_meas)
+
+    measured_parameters = tuple(param for param in param_meas
+                                if isinstance(param, _BaseParameter))
+
+    shapes = get_shape_of_measurement(measured_parameters)
```

Comment:
this is the shape of the _BaseParameter's among param_meas, right?

Reply:
Yes. It is unfortunat that `param_meas` contains functions that are not parameters